tests/test_evpn_l3_vxlan.py

```diff
@@ -0,0 +1,1726 @@
+        create_vlan_interface(dvs, "Vlan100", "Ethernet24", "Vrf-RED", "100.100.3.1/24")
+        vxlan_obj.check_router_interface(dvs, 'Vrf-RED', vxlan_obj.vlan_id_map['100'], 2)
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Add"
+        vxlan_obj.fetch_exist_entries(dvs)
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', "Vlan100", "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Delete"
+        delete_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_del_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTesting IPv4 Route and Overlay Nexthop Update"
+        print "\tTest VRF IPv4 Route with Tunnel Nexthop 7.7.7.7 Add"
+        vxlan_obj.fetch_exist_entries(dvs)
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', "Vlan100", "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest Tunnel Nexthop change from 7.7.7.7 to 8.8.8.8"
+        create_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '8.8.8.8', "Vlan100", "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000', 1)
+        print "\tPass"
+
+        print "\tTest Previous Tunnel Nexthop 7.7.7.7 is removed"
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route and Tunnel Nexthop 8.8.8.8 Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+        delete_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_del_vrf_routes(dvs, "80.80.1.0/24", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTest VRF IPv4 Route with ECMP Tunnel Nexthop Add and Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+
+        ecmp_nexthop_attr = [
+            ("nexthop", "7.7.7.7,8.8.8.8"),
+            ("ifname", "Vlan100,Vlan100"),
+            ("vni_label", "1000,1000"),
+            ("router_mac", "00:11:11:11:11:11,00:22:22:22:22:22"),
+        ]
+
+        print "\tTest VRF IPv4 Route with ECMP Tunnel Nexthop [7.7.7.7 , 8.8.8.8] Add"
+        create_vrf_routes_ecmp(dvs, "80.80.1.0/24", 'Vrf-RED', ecmp_nexthop_attr)
+
+        nh_count = 2
+        ecmp_nhid_list = vxlan_obj.check_vrf_routes_ecmp(dvs, "80.80.1.0/24", 'Vrf-RED', tunnel_name, nh_count)
```

Comment:
This is an ECMP case for the overlay routes right? Is there a way to check both underlay and overlay ecmp paths?

Reply:
Yes this is Overlay route with ECMP Tunnel nexthop test case. 
Routes can have either all overlay nexthop or all underlay nexthops. 
Mix of both overlay and underlay nexthop is currently not supported.